### ISSUE TITLE
What is the recommended solution to mscorlib having a reference to System.Security.Permissions

### ISSUE THREAD

**Issue by vitek-karas** [dotnet/corefx#39221]
I find it really weird that `mscorlib` has an assembly reference (and type forwarders) to an assembly which is not shipping in the `Microsoft.NETCore.App` framework.

I understand that trying to build a new app using  these types results in an error which directs users to install the assembly:
```
Program.cs(10,66): error CS1069: The type name 'DataProtectionPermission' could not be found in the namespace 'System.Security.Permissions'. This type has been forwarded to assembly 'System.Security.Permissions, Version=0.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' Consider adding a reference to that assembly. [E:\AppModel\repro\secper1\secper1.csproj]
```
(Note though that the error says add a reference to an assembly - which doesn't exist, one has to know to go to NuGet and install the package for it - and to pick the right version).

But if I have an existing code, like a library built for .NET Framework which uses these types, there's no way to know this.

What is our guidance to people running into this issue?

**Comment by jkotas** [comment]
> What is our guidance to people running into this issue?

- Run portability analyzer on binaries built for .NET Framework to check for missing APIs: https://docs.microsoft.com/en-us/dotnet/core/porting/

- Add Windows Compatibility Pack to maximimize number of APIs that you have on your disposal: https://devblogs.microsoft.com/dotnet/announcing-the-windows-compatibility-pack-for-net-core/

**Comment by KevinCathcart** [comment]
One issue with pointing people at the portability analyzers is that it uses the API catalogs, which are currently less than optimal with regards to .Net 3.0.

It seems to be including most of the "Windows Compatibility Pack" in the .NET Core 3.0 catalog, rather than the "Platform Extensions" catalog like with 2.x.

I *think* what is going on here is something along the lines that anything exposed via a reference assembly in any of the packs included with the .NET Core 3.0 SDK are being counted as part of .NET Core 3.0.  So System.Security.Permissions is being included because Microsoft.AspNetCore.App.Ref exposes it. Quite a bit of the Windows Compatibility Pack is being exposed in Microsoft.WindowsDesktop.App.Ref. I *suspect* that only those parts of the Windows Compatibility Pack not exposed by any of these are ending up in the "Platform Extensions" catalog for 3.0.

This is less than idea, as it gives the impression that these APIs should be available from a command line 3.0 application, but they are they are not unless the developer does one of the following: change the SDK, add a framework reference to the shared framework that happens to expose this API, or reference the Windows Compatibility Pack package. 

So basically at the moment the Portability Analyzer guidance is not very useful for many of these scenarios.

**Comment by ericstj** [comment]
It's by-design that our desktop compatibility facades (including mscorlib) reference libraries that are not included as part of the default assembly set.  The error you get is actually slightly better than what you'd get if we didn't type forward.  Without the typeforward you simply have a missing type, with it, you have a pointer to an assembly which you can search for.

@carlossanlop / @safern might be able to address the question about the docs.  I'm not aware if the docs process is sophisticated enough to indicate package vs framework reference.
